src/lightning_app/source_code/local.py

```diff
@@ -27,6 +28,14 @@ def __init__(self, path: Path, ignore_functions: Optional[List[_IGNORE_FUNCTION]
         if not self.cache_location.exists():
             self.cache_location.mkdir(parents=True, exist_ok=True)
 
+        # Create a default dotignore if it doesn't exist
```

Comment:
Let's have a test for this new behavior?

Reply:
Was added in #16056 along with a test, just moving it in this PR 😃